paramak/neutronics_utils.py

```diff
@@ -119,3 +124,225 @@ def add_stl_to_moab_core(
+            }
+
+        if tally.name.endswith('spectra'):
+            data_frame = tally.get_pandas_dataframe()
+            tally_result = data_frame["mean"]
+            tally_std_dev = data_frame['std. dev.']
+            results[tally.name]['Flux per source particle'] = {
+                'energy': openmc.mgxs.GROUP_STRUCTURES['CCFE-709'].tolist(),
+                'result': tally_result.tolist(),
+                'std. dev.': tally_std_dev.tolist(),
+            }
+
+        if tally.name.startswith('tritium_production_on_2D_mesh'):
+
+            _save_2d_mesh_tally_as_png(
+                score='(n,Xt)',
+                tally=tally,
+                filename='tritium_production_on_2D_mesh' + tally.name[-3:]
+            )
+
+        if tally.name.startswith('flux_on_2D_mesh'):
+
+            _save_2d_mesh_tally_as_png(
+                score='flux',
+                tally=tally,
+                filename='flux_on_2D_mesh' + tally.name[-3:]
+            )
+
+        if tally.name.startswith('heating_on_2D_mesh'):
+
+            _save_2d_mesh_tally_as_png(
+                score='heating',
+                tally=tally,
+                filename='heating_on_2D_mesh' + tally.name[-3:]
+            )
+
+        if '_on_3D_mesh' in tally.name:
+            mesh_id = 1
+            mesh = statepoint.meshes[mesh_id]
+
+            xs = np.linspace(
+                mesh.lower_left[0],
+                mesh.upper_right[0],
+                mesh.dimension[0] + 1
+            )
+            ys = np.linspace(
+                mesh.lower_left[1],
+                mesh.upper_right[1],
+                mesh.dimension[1] + 1
+            )
+            zs = np.linspace(
+                mesh.lower_left[2],
+                mesh.upper_right[2],
+                mesh.dimension[2] + 1
+            )
+            tally = statepoint.get_tally(name=tally.name)
+
+            data = tally.mean[:, 0, 0]
+            error = tally.std_dev[:, 0, 0]
+
+            data = data.tolist()
+            error = error.tolist()
+
+            for counter, i in enumerate(data):
+                if math.isnan(i):
+                    data[counter] = 0.
+
+            for counter, i in enumerate(error):
+                if math.isnan(i):
+                    error[counter] = 0.
+
+            write_3d_mesh_tally_to_vtk(
+                xs=xs,
+                ys=ys,
+                zs=zs,
+                tally_label=tally.name,
+                tally_data=data,
+                error_data=error,
+                outfile=tally.name + '.vtk'
+            )
+
+    return results
+
+
+def write_3d_mesh_tally_to_vtk(
+        xs,
+        ys,
+        zs,
+        tally_label: str,
+        tally_data,
+        error_data,
+        outfile):
+    try:
+        import vtk
+    except (ImportError, ModuleNotFoundError):
+        msg = "Conversion to VTK requested," \
+            "but the Python VTK module is not installed."
+        raise ImportError(msg)
```

Comment:
These lines aren't tested. But then again, I don't know how to test that a module is not found. Would the native ImportError be self-explanatory ? It may be redundant to re-write the error message ? what do you think ?

Reply:
hummm yes this is trick, I might take this idea to an issue

Reply:
I think it also applies to other places in the code but can't remember. Wonder what the pullrequest review had to say about this

Reply:
We can keep track of this additional testing idea over on issue #679